File: apps/painel_lider/lib/services/auth-service.ts
import { auth, db } from '../firebase';
import {
  signInWithEmailAndPassword,
  createUserWithEmailAndPassword,
  signOut,
  onAuthStateChanged,
  updateProfile,
  User as FirebaseUser
} from 'firebase/auth';
import { doc, setDoc, Timestamp } from 'firebase/firestore';

export async function loginWithEmail(email: string, password: string) {
  try {
    const result = await signInWithEmailAndPassword(auth, email, password);
    return result.user;
  } catch (error) {
    console.error('Erro ao fazer login:', error);
    throw error;
  }
}

export async function registerWithEmail(email: string, password: string) {
  try {
    const result = await createUserWithEmailAndPassword(auth, email, password);
    return result.user;
  } catch (error) {
    console.error('Erro ao registrar:', error);
    throw error;
  }
}

export async function registerMember(name: string, email: string, password: string) {
  const result = await createUserWithEmailAndPassword(auth, email, password);
  await updateProfile(result.user, { displayName: name });
  await setDoc(doc(db, 'usuarias', result.user.uid), {
    nome: name,
    email,
    papel: 'membro',
    ultimoAcesso: Timestamp.now(),
  });
  return result.user;
}

export async function logout() {
  try {
    await signOut(auth);
  } catch (error) {
    console.error('Erro ao fazer logout:', error);
    throw error;
  }
}

export function onAuthChange(callback: (user: FirebaseUser | null) => void) {
  return onAuthStateChanged(auth, callback);
}

File: apps/app_mobile/lib/core/services/auth_service.dart
import 'package:firebase_auth/firebase_auth.dart';
import 'package:cloud_firestore/cloud_firestore.dart';

class AuthService {
  static final _auth = FirebaseAuth.instance;
  static final _db = FirebaseFirestore.instance;

  static User? get currentUser => _auth.currentUser;
  static Stream<User?> get authState => _auth.authStateChanges();

  static Future<void> signIn(String email, String password) async {
    await _auth.signInWithEmailAndPassword(
      email: email.trim(),
      password: password,
    );
    await _updateLastAccess();
  }

  static Future<void> signUp(String name, String email, String password) async {
    final cred = await _auth.createUserWithEmailAndPassword(
      email: email.trim(),
      password: password,
    );
    await cred.user?.updateDisplayName(name.trim());
    await _db.collection('usuarias').doc(cred.user!.uid).set({
      'nome': name.trim(),
      'email': email.trim(),
      'papel': 'membro',
      'criadoEm': FieldValue.serverTimestamp(),
      'ultimoAcesso': FieldValue.serverTimestamp(),
    });
  }

  static Future<void> signOut() => _auth.signOut();

  static Future<void> _updateLastAccess() async {
    final uid = _auth.currentUser?.uid;
    if (uid == null) return;
    await _db.collection('usuarias').doc(uid).set(
      {'ultimoAcesso': FieldValue.serverTimestamp()},
      SetOptions(merge: true),
    );
  }
}
